src/utilities.py

```diff
@@ -411,18 +411,19 @@ def break_long_message(phrases, joinstr = " "):
 
 #completes a partial nickname or string from a list
 def complete_match(string, matches):
-    num_matches = 0
-    bestmatch = string
+    possible_matches = set()
     for possible in matches:
         if string == possible:
-            return string, 1
+            return [string]
         if possible.startswith(string) or possible.lstrip("[{\\^_`|}]").startswith(string):
-            bestmatch = possible
-            num_matches += 1
-    if num_matches != 1:
-        return None, num_matches
-    else:
-        return bestmatch, 1
+            possible_matches.add(possible)
+    return sorted(possible_matches)
+
+def complete_one_match(string, matches):
+    match = complete_match(string,matches) 
+    if len(match) == 1:
+        return match.pop()
+    return None
```

Comment:
Not needed in practice (Python adds an implicit `return None` at the end of every function), but you could keep it as a "Returning None is intended" hint.